runelite-client/src/main/java/net/runelite/client/plugins/config/PluginSearch.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020, Jack Hodkinson <https://github.com/jrhodkinson>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.runelite.client.util.Text;
+import static java.util.stream.Collectors.toList;
+
+public class PluginSearch<T>
+{
+
+	private final List<SearchablePlugin<T>> searchablePlugins;
+
+	public PluginSearch(List<SearchablePlugin<T>> searchablePlugins)
+	{
+		this.searchablePlugins = new ArrayList<>(searchablePlugins);
+	}
+
+	public List<T> query(String query)
```

Comment:
this can probably just be a static method on the comparator

Reply:
Decided to merge the old `PluginSearch` and `PluginSearchComparator`, making this a static method and making the comparator constructor private